### ISSUE TITLE
MappingException when removing based on query using lt

### ISSUE THREAD

**Issue by anz000** [kaiso/relmongo#48]
I was working on relmongo `v3.0.0`. Recently, I realized that there were newer versions avaialble, so I decided to upgrade to `v3.2.0`. However, one of the test cases that was passing previously started to fail on me.

Code
```
Criteria expression = Criteria.where("expirationDate").lt(LocalDateTime.now());
Query query = new Query().addCriteria(expression);
return mongoTemplate.remove(query, UserPasswordReset.class).wasAcknowledged();
```
The above code was/is working with `v3.0.0`., but not with versions `greater than v3.0.0.` (v3.0.1, v3.1.0 and v3.2.0). 

[Note :: I played around with other `mongoTemplate` methods like `find` and it seems to work with all versions. Also, I have similar codes using query and criterion that is based on `String` and `remove` method, and those are working without any issues on all versions.]

The associated class
```
@Getter
@ToString
@Document(collection = "passwordResets")
@EntityListeners(AuditingEntityListener.class)
public class UserPasswordReset {

    @Id
    private ObjectId id;

    @NotNull
    private String email;

    private String token = UUID.randomUUID().toString();

    private LocalDateTime expirationDate = LocalDateTime.now();

    @CreatedDate
    private LocalDateTime createdDate;

    // overloaded constructor
    public UserPasswordReset(String email){ this.email = email; }
}
```

Error message
```
org.springframework.data.mapping.MappingException: Expected to read Document Document{{$lt=Mon Mar 16 14:30:08 CDT 2020}} into type class java.time.LocalDateTime but didn't find a PersistentEntity for the latter!

	at org.springframework.data.mongodb.core.convert.MappingMongoConverter.read(MappingMongoConverter.java:244)
	at org.springframework.data.mongodb.core.convert.MappingMongoConverter.readValue(MappingMongoConverter.java:1493)
	at org.springframework.data.mongodb.core.convert.MappingMongoConverter$MongoDbPropertyValueProvider.getPropertyValue(MappingMongoConverter.java:1391)
	at org.springframework.data.mongodb.core.convert.MappingMongoConverter.readProperties(MappingMongoConverter.java:380)
	at org.springframework.data.mongodb.core.convert.MappingMongoConverter.populateProperties(MappingMongoConverter.java:297)
	at org.springframework.data.mongodb.core.convert.MappingMongoConverter.read(MappingMongoConverter.java:277)
	at org.springframework.data.mongodb.core.convert.MappingMongoConverter.read(MappingMongoConverter.java:247)
	at org.springframework.data.mongodb.core.convert.MappingMongoConverter.read(MappingMongoConverter.java:196)
	at org.springframework.data.mongodb.core.convert.MappingMongoConverter.read(MappingMongoConverter.java:192)
	at org.springframework.data.mongodb.core.convert.MappingMongoConverter.read(MappingMongoConverter.java:80)
	at io.github.kaiso.relmongo.events.callback.PersistentPropertyCascadingRemoveCallback.<init>(PersistentPropertyCascadingRemoveCallback.java:50)
	at io.github.kaiso.relmongo.events.processor.RelMongoProcessor.onBeforeDelete(RelMongoProcessor.java:110)
	at org.springframework.data.mongodb.core.mapping.event.AbstractMongoEventListener.onApplicationEvent(AbstractMongoEventListener.java:68)
	at org.springframework.data.mongodb.core.mapping.event.AbstractMongoEventListener.onApplicationEvent(AbstractMongoEventListener.java:31)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:402)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:359)
	at org.springframework.data.mongodb.core.MongoTemplate.maybeEmitEvent(MongoTemplate.java:2276)
	at org.springframework.data.mongodb.core.MongoTemplate$9.doInCollection(MongoTemplate.java:1694)
	at org.springframework.data.mongodb.core.MongoTemplate$9.doInCollection(MongoTemplate.java:1689)
	at org.springframework.data.mongodb.core.MongoTemplate.execute(MongoTemplate.java:545)
	at org.springframework.data.mongodb.core.MongoTemplate.doRemove(MongoTemplate.java:1689)
	at org.springframework.data.mongodb.core.MongoTemplate.remove(MongoTemplate.java:1677)
	at org.springframework.data.mongodb.core.MongoTemplate.remove(MongoTemplate.java:1670)
	at com.timecloud.user.repository.UserPasswordResetCustomRepositoryImpl.deleteAllInactive(UserPasswordResetCustomRepositoryImpl.java:45)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	
```

**Comment by kaiso** [comment]
Hello @anz000 Thank you for reporting the problem, I will test on my side and get back to you

**Comment by kaiso** [comment]
@anz000 i've released the version **v3.2.1** with a fix for the problem, please test and give me your feedback.

**Comment by anz000** [comment]
My test cases passes with v3.2.1. Thanks.
